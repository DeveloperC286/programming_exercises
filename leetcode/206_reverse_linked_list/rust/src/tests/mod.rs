use insta::assert_debug_snapshot;

use super::*;

#[test]
fn test_empty_list() {
    // Given
    let list = None;

    // When
    let actual = reverse_list(list);

    // Then
    assert_debug_snapshot!("test_empty_list", actual);
}

#[test]
fn test_single_node_list() {
    // Given
    let list = Some(Box::new(ListNode { val: 1, next: None }));

    // When
    let actual = reverse_list(list);

    // Then
    assert_debug_snapshot!("test_single_node_list", actual);
}

#[test]
fn test_odd_length_list() {
    // Given
    let list = Some(Box::new(ListNode {
        val: 1,
        next: Some(Box::new(ListNode {
            val: 2,
            next: Some(Box::new(ListNode { val: 3, next: None })),
        })),
    }));

    // When
    let actual = reverse_list(list);

    // Then
    assert_debug_snapshot!("test_odd_length_list", actual);
}
