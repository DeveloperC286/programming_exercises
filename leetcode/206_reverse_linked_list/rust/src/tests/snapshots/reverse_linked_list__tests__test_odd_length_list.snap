---
source: src/tests/mod.rs
expression: actual

---
Some(
    ListNode {
        val: 3,
        next: Some(
            ListNode {
                val: 2,
                next: Some(
                    ListNode {
                        val: 1,
                        next: None,
                    },
                ),
            },
        ),
    },
)
